connector/oidc/oidc.go

```diff
@@ -56,16 +56,24 @@ type Config struct {
 	// PromptType will be used fot the prompt parameter (when offline_access, by default prompt=consent)
 	PromptType string `json:"promptType"`
 
-	ClaimMapping struct {
-		// Configurable key which contains the preferred username claims
-		PreferredUsernameKey string `json:"preferred_username"` // defaults to "preferred_username"
+	// OverrideClaimMapping will be used to override the options defined in claimMappings.
+	// i.e. if there are 'email' and `preferred_email` claims available, by default Dex will always use the `email` claim independent of the ClaimMapping.EmailKey.
+	// This setting allows you to override the default behavior of Dex and enforce the mappings defined in `claimMapping`.
+	OverrideClaimMapping bool `json:"overrideClaimMapping"` // defaults to false
 
-		// Configurable key which contains the email claims
-		EmailKey string `json:"email"` // defaults to "email"
+	ClaimMapping ClaimMapping `json:"claimMapping"`
+}
+
+type ClaimMapping struct {
```

Comment:
Can you please revert the changes that are not relevant to this feature? An inline struct might not be the nicest solution, but it should disappear eventually anyway.

If (for some reason) this is an important change, can you please make it a separate commit so that the refactor clearly separates from the feature?

Reply:
Done